packages/app-desktop/gui/Sidebar/Sidebar.tsx

```diff
@@ -744,7 +744,7 @@ const SidebarComponent = (props: Props) => {
 
 
 	if (props.folders.length) {
-		const allNotesSelected = props.selectedFolderId === ALL_NOTES_FILTER_ID;
+		const allNotesSelected = props.selectedFolderId === null ? true : props.selectedFolderId === ALL_NOTES_FILTER_ID;
```

Comment:
Does this mean that "All notes" can have as an ID either `ALL_NOTES_FILTER_ID` or `null`? And isn't it possible to ensure it always has `ALL_NOTES_FILTER_ID` as ID?

Reply:
I think it's because of this existing check: 
```
// Sidebar.tsx
const folderItem_click = useCallback((folderId: string) => {
        props.dispatch({
	        type: 'FOLDER_SELECT',
	        id: folderId ? folderId : null,
        });
}, [props.dispatch]);
```
But maybe AliceHincu found some issues or part of the codebase that I am unaware of.

Reply:
I understand now. It's because when selecting a `SmartFilter,` the `selectedSmartFilterId` is updated, not 
 the`selectedFolderId`. And because now in AliceHincu, switch to the `SmartFilter` folder type later, then the `selectedFolderId` is null.

I suggest this: `props.notesParentType === 'SmartFilter' && props.selectedSmartFilterId === ALL_NOTES_FILTER_ID` and create a comment that `SmartFilter` folder types do not use `selectedFolderId` but `selectedSmartFilterId`

Reply:
Yes that seems like a reasonable solution for now.

Reply:
Hi! I will implement these changes today and I will also check if there are any problems related to the change. I will be back with updates.

Reply:
Oh, I remember now why I checked for null. So I actually did see that there is a `notesParentType` attribute, but I noticed that it always had the value `Folder`. The proposed solution is insufficient on its own. Here is a full investigation:

Inside the `buildUi` method, the folderList is initialized like this:

`const folderList = new FolderListWidget();`
 
 Inside the FolderListWidget's constructor, there is the prop initialized:
 
 `this.notesParentType_ = 'Folder'; `
 
This explains why, at first, the application will behave strangely even though the user interface (UI) indicates that all notes are chosen. Instead, it will consider the "All notes" section as a folder.. I replaced `this.notesParentType_ = 'Folder';` with `this.notesParentType_ = 'SmartFilter';` and it works. (instead of having the initial value _Folder_, we can set the initial value to _SmartFilter_)

Reply:
Also with this modification, I have checked and we don't have to call `folderItem_click(ALL_NOTES_FILTER_ID);` anymore.

Reply:
I updated the code. Also, I have noticed that we don't need the call to `folderItem_click` anymore, the code works fine without it... I also noticed that in tests, when wanting to test the _All Notes_ section, dispatching the action SMART_FILTER_SELECT is enough. I do not know if I am missing something, since I can't see how it can be useful for us anymore. I switched between having the call and not having the call and I don't see any differences.

Reply:
Basically, I added `folderItem_click` for the feature where you can have custom ordering for notes in `All notes`. However, I think that is the out of the scope of this PR. I suggest that I would create another PR for that, let's see what laurent22 has to say.

Anyway, in my opinion the code looks good for now :)